zplugin-ros1/src/ros_to_zenoh_bridge/aloha_subscription.rs

```diff
@@ -0,0 +1,276 @@
+        F: Fn(
+                zenoh::key_expr::KeyExpr,
+            ) -> Box<dyn futures::Future<Output = ()> + Unpin + Send + Sync>
+            + Send
+            + Sync
+            + 'static,
+    {
+        let task_running = Arc::new(AtomicBool::new(true));
+
+        async_std::task::spawn(AlohaSubscription::task(
+            task_running.clone(),
+            key,
+            beacon_period,
+            session,
+            on_resource_declared,
+            on_resource_undeclared,
+        ));
+
+        Ok(Self { task_running })
+    }
+
+    //PRIVATE:
+    async fn task<F>(
+        task_running: Arc<AtomicBool>,
+        key: OwnedKeyExpr,
+        beacon_period: Duration,
+        session: Arc<Session>,
+        on_resource_declared: F,
+        on_resource_undeclared: F,
+    ) -> ZResult<()>
+    where
+        F: Fn(
+                zenoh::key_expr::KeyExpr,
+            ) -> Box<dyn futures::Future<Output = ()> + Unpin + Send + Sync>
+            + Send
+            + Sync
+            + 'static,
+    {
+        let mut accumulating_resources = Cell::new(BTreeMap::<String, AlohaResource>::new());
+        let subscriber = session.declare_subscriber(key).res_async().await?;
+
+        Self::accumulating_task(
+            task_running,
+            beacon_period * 3,
+            &mut accumulating_resources,
+            &subscriber,
+            on_resource_declared,
+            on_resource_undeclared,
+        )
+        .await;
+
+        Ok(())
+    }
+
+    async fn listening_task<'a, F>(
+        task_running: Arc<AtomicBool>,
+        accumulating_resources: &mut Cell<BTreeMap<String, AlohaResource>>,
+        subscriber: &'a zenoh::subscriber::Subscriber<'a, Receiver<Sample>>,
+        on_resource_declared: &F,
+    ) where
+        F: Fn(
+                zenoh::key_expr::KeyExpr,
+            ) -> Box<dyn futures::Future<Output = ()> + Unpin + Send + Sync>
+            + Send
+            + Sync
+            + 'static,
+    {
+        while task_running.load(Relaxed) {
+            match subscriber.recv_async().await {
+                Ok(val) => {
+                    match accumulating_resources
+                        .get_mut()
+                        .entry(val.key_expr.to_string())
+                    {
+                        Occupied(mut val) => {
+                            val.get_mut().update();
+                        }
+                        Vacant(entry) => {
+                            entry.insert(AlohaResource::new());
+                            on_resource_declared(val.key_expr).await;
+                        }
+                    }
+                }
+                Err(e) => {
+                    error!("Listening error: {}", e);
+                }
+            }
+        }
+    }
+
+    async fn accumulating_task<'a, F>(
+        task_running: Arc<AtomicBool>,
+        accumulate_period: Duration,
+        accumulating_resources: &mut Cell<BTreeMap<String, AlohaResource>>,
```

Comment:
Based on line `209`, `String` must be a valid `keyexpr`. Why not reflecting this by the correct type in the `BTreeMap`? I.e., replacing `String` with `OwnedKeyExpr`? By doing so, there is no need of `unsafe` code at line `209`.